airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -688,19 +687,16 @@ def check_status(
             except ClientError:
                 raise AirflowException("AWS request failed, check logs for more info")
 
-            if status in non_terminal_states:
-                running = True
-            elif status in self.failed_states:
+            if status in self.failed_states:
                 raise AirflowException(f"SageMaker job failed because {response['FailureReason']}")
-            else:
-                running = False
+            elif status not in non_terminal_states:
+                break
 
             if max_ingestion_time and sec > max_ingestion_time:
                 # ensure that the job gets killed if the max ingestion time is exceeded
                 raise AirflowException(f"SageMaker job took more than {max_ingestion_time} seconds")
 
         self.log.info("SageMaker Job completed")
-        response = describe_function(job_name)
```

Comment:
there is basically no latency between the last call of the loop (l.682) and this one, and moreover the status reached is supposed to be a terminal state, so there is no need to repeat the call here.